hawc/apps/common/autocomplete.py

```diff
@@ -0,0 +1,60 @@
+from dal import autocomplete
+from django.http import Http404
+from django.utils.encoding import force_str
+
+from .helper import reverse_with_query_lazy
+
+
+class BaseAutocomplete(autocomplete.Select2QuerySetView):
+    # AJAX only? `request.is_ajax()`
+    # Other permissions?
```

Comment:
i think we should by default require user is authenticated too, though i think in some cases we may need to add additional permissions